07-errors.md

```diff
@@ -0,0 +1,301 @@
+# Error Conditions
+
+This document specifies the error conditions in the FVM and the associated error codes/numbers.
+
+## Exit Codes
+
+Exit codes are numeric, 32bit integer codes that are returned when an actor exits both successfully and in error.
+Exit codes appear on-chain inside the message receipt, and therefore form part of consensus.
+
+We classify exit codes as "normal" and "abnormal".
+
+The only _normal_ exit code is `0`, or `Ok`. The `Ok` exit code indicates that the message was
+applied successfully.
+
+All other exit codes are _abnormal_ and indicate that the message _was not_ applied and that the
+chain state _was not_ updated.
+
+### System "Abnormal Exit" Codes
+
+System abnormal exit codes are generated by the VM itself to indicate that the message itself failed
+to apply. These exit codes will never be observed or returned directly by actors.
+
+| Code  | Status                     | Description                                               |
+|-------|----------------------------|-----------------------------------------------------------|
+| 1     | `SysErrSenderInvalid`      | sender doesn't exist                                      |
+| 2     | `SysErrSenderStateInvalid` | sender's message shouldn't have been included             |
+| 3     | reserved                   |                                                           |
+| 4     | `SysErrActorPanic`         | message receiver trapped                                  |
+| 5     | `SysErrInvalidReceiver`    | receiver doesn't exist and can't be automatically created |
+| 6     | `SysErrInsufficientFunds`  | sender didn't have the requisite funds                    |
+| 7     | `SysErrOutOfGas`           | out of gas                                                |
+| 8     | reserved                   |                                                           |
+| 9     | `SysErrIllegalActor`       | message receiver made an unrecoverable illegal syscall    |
```

Comment:
```suggestion
| 9     | `SysErrIllegalActor`       | message receiver made an unrecoverable illegal syscall    |
```

Illegal actor to me implies something about the actor _is_ illegal, not that the actor _did_ something illegal. If this error code is only raised on illegal calls, then let's rename to `SysErrIllegalActorCall`.

Reply:
Really, this can _only_ be triggered by calling `vm::abort` with a bad exit code. Maybe `SysErrIllegalExitCode`?

Reply:
Ah, yes, if it can only be triggered in that case, then let's be more specific. `SysErrIllegalExitCode` works for me.